src/main/java/org/chrisle/netbeans/plugins/nbscratchfile/CreateScratchFile.java

```diff
@@ -1,158 +1,142 @@
-                        colorizeElement((Element) webEngine.getDocument().getElementsByTagName("ul").item(0), String.format("color: %s;", viewModel.getColor("Label.foreground", false)));
-
-                        addHoverEffectToElements(webEngine.getDocument().getElementsByTagName("li"), String.format("background-color: %s; color: %s;", viewModel.getColor("Menu.background", true), viewModel.getColor("Menu.foreground", true)), String.format("background-color: %s; color: %s;", viewModel.getColor("Menu.background", false), viewModel.getColor("Menu.foreground", false)));
-                    }
-                });
-            } catch (Exception ex) {
-                Exceptions.printStackTrace(ex);
-            }
-        });
-    }
-
-    public void showDialog() {
-        // try to use monitor, where the input focus is
-        // therefor get the topmost component based on the input focus
-        Component focusOwner = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
-
-        if (null != focusOwner) {
-            while (focusOwner.getParent() != null) {
-                focusOwner = focusOwner.getParent();
-            }
-        }
-
-        dialog.setLocationRelativeTo(focusOwner);
-        dialog.setVisible(true);
-    }
+package org.chrisle.netbeans.plugins.nbscratchfile;
+
+import java.awt.Component;
+import java.awt.KeyboardFocusManager;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowFocusListener;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.KeyStroke;
+import net.java.html.js.JavaScriptBody;
+import org.netbeans.api.htmlui.HTMLComponent;
+import org.netbeans.api.htmlui.HTMLDialog;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.NbBundle.Messages;
+
+@ActionID(
+        category = "Tools",
+        id = "org.chrisle.netbeans.plugins.nbscratchfile.CreateScratchFile"
+)
+@ActionRegistration(
+        displayName = "#CTL_CreateScratchFile",
+        iconBase = "org/chrisle/netbeans/plugins/nbscratchfile/add_file.png"
+)
+@ActionReferences({
+    @ActionReference(path = "Menu/File", position = 150)
+    ,
+    @ActionReference(path = "Shortcuts", name = "DOS-N")
+})
+@Messages("CTL_CreateScratchFile=New Scratch File...")
+public final class CreateScratchFile implements ActionListener {
+    private final JDialog dialog;
+    private final JComponent jfxPanel;
+    private final NbScratchFileViewModel viewModel;
+
+    public CreateScratchFile() {
+        dialog = new JDialog();
+        viewModel = new NbScratchFileViewModel(dialog);
+        jfxPanel = Pages.initWebView(viewModel);
```

Comment:
This call hides the initialization of JavaFX `WebView` currently. However it shall be possible to replace it with other web view implementation in the future.